ex2_services/Services/source/communication/communication_service.c

```diff
@@ -756,6 +756,31 @@ SAT_returnState communication_service_app(csp_packet_t *packet) {
 
         break;
     }
+    case UHF_GET_SWVER: {
+        uint16_t version;
+        status = HAL_UHF_getSWVersion(&version);
+        if (sizeof(version) + 1 > csp_buffer_data_size()) {
+            return_state = SATR_ERROR;
+        }
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        memcpy(&packet->data[OUT_DATA_BYTE], &version, sizeof(version));
+        set_packet_length(packet, sizeof(int8_t) + sizeof(version) + 1);
```

Comment:
I wonder if we might want a SET_RESPONSE macro that sets the status, response and packet length? I guess it might seem like too late now ;->

Reply:
Ha yes that would have been smart